libphal/phal_dump.C

```diff
@@ -169,473 +219,574 @@ void writeSbeData(const std::filesystem::path& dumpPath, uint32_t reasonCode,
+ * @param[in] failingUnit Position of the proc containing the failed SBE
+ * @param[in] sbeTypeId The chip type number
  *
- * @throw std::runtime_error Throws if there is an error in collecting the PIBMS
- * register dump or writing to the file.
+ * Exceptions: PDBG_INIT_FAIL for any pdbg init related failure.
+ *             PDBG_TARGET_NOT_OPERATIONAL for invalid failing unit
+ *             HWP_EXECUTION_FAILED if the extract rc procedure is failing
  */
-void collectPIBMSRegDump(struct pdbg_target* proc,
-			 const std::filesystem::path& dumpPath,
-			 const std::string& baseFilename)
+struct pdbg_target* preCollection(const uint32_t failingUnit,
+				  const std::filesystem::path& dumpPath,
+				  const int sbeTypeId)
 {
-	std::vector<sRegV> pibmsRegSet;
-	for (auto& reg : pibms_regs_2dump) {
-		sRegV regv;
-		regv.reg = reg;
-		pibmsRegSet.emplace_back(regv);
-	}
-
-	ReturnCode fapiRc = p10_pibms_reg_dump(proc, pibmsRegSet);
-	if (fapiRc != FAPI2_RC_SUCCESS) {
-		log(level::ERROR,
-		    "Failed in p10_pibms_reg_dump for proc=%s, rc=0x%08X",
-		    pdbg_target_path(proc), fapiRc);
-		throw std::runtime_error("p10_pibms_reg_dump failed");
-	}
-	std::vector<DumpPIBMSRegVal> dumpRegs;
-	for (auto& reg : pibmsRegSet) {
-		dumpRegs.emplace_back(reg.reg.addr, reg.reg.name, reg.reg.attr,
-				      reg.value);
-	}
-
-	std::string dumpFilename = baseFilename + "p10_pibms_reg_dump";
-	std::filesystem::path basePath = dumpPath / dumpFilename;
-
-	// Writing the PIBMS register values to a file
-	writePIBMSRegValuesToFile(dumpRegs, basePath);
+	initializePdbgLibEkb();
+	// Find the proc target from the failing unit id
+	auto chip = getTargetFromFailingId(failingUnit, sbeTypeId);
+	// To check SBE state P10 uses a pib target which in PDBG
+	// library internally gets translated into fsi one
+	// But for Odyssey we directly pass on a fsi target
+	struct pdbg_target* pib_fsi = nullptr;
+	if (PROC_SBE_DUMP == sbeTypeId)
+		pib_fsi = probePibTarget(chip, sbeTypeId);
+	else if (ODYSSEY_SBE_DUMP == sbeTypeId)
+		pib_fsi = probeFsiTarget(chip, sbeTypeId);
+
+	checkSbeState(pib_fsi, sbeTypeId);
+	extractSbeRc(chip, dumpPath, sbeTypeId);
+	return chip;
 }
 
 /**
- * @brief Writes PIBMEM data to a file.
- * @param[in] dumpData A vector of uint64_t values representing the PIBMEM data
- * to be written.
- * @param[in] basePath The filesystem path of the file to which the data will be
- * written.
+ * @brief Write the dump file to the path
+ * @param[in] data Pointer to the data buffer
+ * @param[in] len Length of the buffer
+ * @param[in] dumpPath Path of the file
  *
- * @throw std::runtime_error Throws if there is an error opening the file or
- * during the file write operation.
+ * Throws FILE_OPERATION_FAILED in the case of error
  */
-void writePIBMEMDataToFile(const std::vector<uint64_t>& dumpData,
-			   const std::filesystem::path& basePath)
+void writeDumpFile(char* data, size_t len, std::filesystem::path& dumpPath)
```

Comment:
this function was removed

Reply:
Rather than having four different functions for writing four different HWPs dump contents with same lines of code and logic I have tried to optimize it into a single function which will handle the dump contents from the all four HWPs, thus removed.

Reply:
optimization in a different pull request with a different story

Reply:
This function `writeDumpFile` is not been changed actually. Only it's place has been changed due to introduction of other functions above.